pkg/cmd/container/create.go

```diff
@@ -357,8 +365,49 @@ func generateRootfsOpts(args []string, id string, ensured *imgutil.EnsuredImage,
 			// error message is from Podman
 			return nil, nil, errors.New("no command or entrypoint provided, and no CMD or ENTRYPOINT from image")
 		}
+
+		entrypointPath = processArgs[0]
+
 		opts = append(opts, oci.WithProcessArgs(processArgs...))
 	}
+
+	systemdPaths := []string{
+		"/usr/sbin/init",
+		"/sbin/init",
+		"/usr/local/sbin/init",
+	}
+
+	stopSignal := options.StopSignal
+
+	if options.Systemd == "always" || (options.Systemd == "true" && slices.Contains(systemdPaths, entrypointPath)) {
+		if options.Privileged {
+			securityOptsMap := strutil.ConvertKVStringsToMap(strutil.DedupeStrSlice(options.SecurityOpt))
+			privilegedWithoutHostDevices, err := maputil.MapBoolValueAsOpt(securityOptsMap, "privileged-without-host-devices")
+			if err != nil {
+				return nil, nil, err
+			}
+
+			// See: https://github.com/containers/podman/issues/15878
+			if !privilegedWithoutHostDevices {
+				return nil, nil, errors.New("If --privileged is used with systemd `--security-opt privileged-without-host-devices` must also be used")
```

Comment:
In the `podman` implementation all `/dev/tty*` devices are unmounted to prevent causing host to crash

I could not find an easy to achieve this, so instead I return an error (also prevents causing host to crash)